torch/jit/_monkeytype_config.py

```diff
@@ -0,0 +1,124 @@
+    print("Warning: monkeytype is not installed. Please install https://github.com/Instagram/MonkeyType "
+          "to enable Profile-Directed Typing in TorchScript. Refer https://github.com/Instagram/MonkeyType/blob/master/README.rst "
+          "to install MonkeyType. ")
+
+def get_qualified_name(func):
+    # For class methods, the method name also includes class name. To
+    # To extract the method name, we split the function name using "."
+    # The last value in the list provides the method name.
+    return func.__qualname__.split(".")[-1]
+
+if _IS_MONKEYTYPE_INSTALLED:
+
+    class JitTypeTraceStoreLogger(CallTraceStoreLogger):
+        """A JitTypeCallTraceLogger that stores logged traces in a CallTraceStore."""
+        def __init__(self, store: CallTraceStore):
+            super().__init__(store)
+
+        def log(self, trace: CallTrace) -> None:
+            self.traces.append(trace)
+
+    class JitTypeTraceStore(CallTraceStore):
+        def __init__(self):
+            super().__init__()
+            # A dictionary keeping all collected CallTrace
+            # key is fully qualified name of called function
+            # value is list of all CallTrace
+            self.trace_records: Dict[str, list] = defaultdict(list)
+
+        def add(self, traces: Iterable[CallTrace]):
+            for t in traces:
+                qualified_name = get_qualified_name(t.func)
+                self.trace_records[qualified_name].append(t)
+
+        def filter(
+            self,
+            qualified_name: str,
+            qualname_prefix: Optional[str] = None,
+            limit: int = 2000
+        ) -> List[CallTraceThunk]:
+            return self.trace_records[qualified_name]
+
+        def analyze(self, qualified_name: str) -> Dict:
+            # Analyze the types for the given module
+            # and create a dictionary of all the types
+            # for arguments.
+            records = self.trace_records[qualified_name]
+            all_args = defaultdict(set)  # type:  ignore[var-annotated]
+            for record in records:
+                for arg, arg_type in record.arg_types.items():
+                    all_args[arg].add(arg_type)
+            return all_args
+
+        def consolidate_types(self, qualified_name: str) -> Dict:
+            all_args = self.analyze(qualified_name)
+            # If there are more types for an argument,
+            # then consolidate the type to `Any` and replace the entry
+            # by type `Any`.
+            for arg, types in all_args.items():
+                _all_type = " "
+                for _type in types:
+                    # If the type is a type imported from typing
+                    # like Tuple, List, Dict then replace "typing."
+                    # with a null string.
+                    _type_to_string = str(_type)
+                    if "typing" in _type_to_string:
+                        _all_type += _type_to_string.replace("typing.", '') + ","
```

Comment:
Is this str replacement necessary? Here we are changing `typing.List` to `List`, which I think is why you had to do this `from typing import List` (and potentially other containers like `Dict`, `Tuple`) in the other file. If we don't remove `typing` from the string, we can just import entire `typing` module to wherever the annotations are parsed.

Reply:
Gave this a quick try. If I retain the `typing`, then I need to import the entire `typing` module in `test_pdt.py`.  Tried adding `import typing` in `frontend.py`/ `annotations.py`. That did not work.

Reply:
Strange. That sounds like an orthogonal issue though, I am OK with leaving it as is for now.